[integration-cli] [--net none] Fix TestPullImageOfficialNames

Fix TestPullImageOfficialNames so it will still be a functional test, but will pass when the test container has --net none.

Use the local registry in the container to pull an image. 

Error output when tests are run with --net none:
```
--- FAIL: TestPullImageOfficialNames (0.32s)
	docker_cli_pull_test.go:120: pulling the 'docker.io/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'index.docker.io/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'library/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'docker.io/library/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'index.docker.io/library/hello-world' image from the registry has failed: exit status 1

```

--- a/integration-cli/docker_cli_pull_test.go
+++ b/integration-cli/docker_cli_pull_test.go
@@ -108,6 +108,8 @@ func TestPullNonExistingImage(t *testing.T) {
 // pulling an image from the central registry using official names should work
 // ensure all pulls result in the same image
 func TestPullImageOfficialNames(t *testing.T) {
+	testRequires(t, Network)
+
 	names := []string{
 		"docker.io/hello-world",
 		"index.docker.io/hello-world",

--- a/integration-cli/requirements.go
+++ b/integration-cli/requirements.go
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"net/http"
 	"os/exec"
 	"strings"
 	"testing"
@@ -31,6 +32,16 @@ var (
 	ExecSupport = TestRequirement{
 		func() bool { return supportsExec },
 		"Test requires 'docker exec' capabilities on the tested daemon.",
+	}
+	Network = TestRequirement{
+		func() bool {
+			resp, err := http.Get("http://hub.docker.com")
+			if resp != nil {
+				resp.Body.Close()
+			}
+			return err == nil
+		},
+		"Test requires network availability, environment variable set to none to run in a non-network enabled mode.",
 	}
 	RegistryHosting = TestRequirement{
 		func() bool {

